samples/basic/secret.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp
+from kfp import dsl
+from kfp.gcp import use_gcp_secret
+
+
+def gcs_read_op(url):
+  return dsl.ContainerOp(
+      name='Access GCS using auth token',
+      image='google/cloud-sdk:latest',
+      command=['sh', '-c'],
+      arguments=['gsutil ls ' + str(url)]
+      )
+
+
+def use_secret_json_op():
+  return dsl.ContainerOp(
+      name='view auth token location',
+      image='google/cloud-sdk:latest',
+      command=['sh', '-c'],
+      arguments=[
+          'gcloud auth activate-service-account --key-file /secret/gcp-credentials/user-gcp-sa.json'
+          ])
+
+
+@dsl.pipeline(
+    name='Secret pipeline',
+    description='A pipeline to demonstrate mounting and use of secretes.'
+)
+
```

Comment:
Usually there is no space between the decorator and the decorated function.
JFYI, the name and description are now optional. And the decorator itself is only needed for the command-line `dsl-compile`.

Reply:
Thank you, updated accordingly.

Reply:
Regarding name and description, is there a plan to remove this field completely? seems like it would makes sense to either require it in both places or remove it from both.

Reply:
The `@pipeline` decorator is required when compiling the pipeline using `dsl-compile`.
The `@pipeline` decorator can be used to override the pipeline name (e.g. provide a human-readable name instead of the pythonic function name).